src/headless/converse-core.js

```diff
@@ -1272,6 +1272,9 @@ _converse.initialize = async function (settings, callback) {
             const password = _.isNil(credentials)
                 ? _converse.connection.pass || this.password
                 : credentials.password;
+            const jid = _.isNil(credentials)
+                ? _converse.connection.authcid || this.jid
+                : credentials.jid;
```

Comment:
Why do you add `_converse.connection.authcid` there as well?

Reply:
@jcbrand I am quite new to Strophe and I tried to make my code similar to password obtaining process (from Strophe connection). I can just use 
```javascript
const jid = _.isNil(credentials)
                ? this.jid
                : credentials.jid;
```
Is it a better option?

Reply:
Yes, I don't think authcid is necessary

Reply:
@jcbrand done. But tests are failing for some strange reason